tests/functional/test_build_env.py

```diff
@@ -42,7 +42,7 @@ def run_with_build_env(
 
             link_collector = LinkCollector(
                 session=PipSession(),
-                search_scope=SearchScope.create([{scratch!r}], [], False),
+                search_scope=SearchScope.create([{str(script.scratch_path)!r}], [], False),
```

Comment:
A local `scratch_path` variable would avoid the E501.

Reply:
Fixed as requested